### PR TITLE
Fix #8921: Command `az keyvault key/secret/certificate list/list-deleted/list-versions` with parameter `--maxresults` do not work as intended

### LINKED ISSUE #8921: Command `az keyvault key/secret/certificate list/list-deleted/list-versions` with parameter `--maxresults` do not work as intended
**Describe the bug**
`az keyvault secret list` has an optional argument that specifies the maximum number of results to return. When a user specifies this argument, they receive an error `Parameter 'maxresults' failed to meet validation requirement.`

**To Reproduce**
$ `az keyvault secret list --vault-name my-keyvault --maxresults 1000`
Parameter 'maxresults' failed to meet validation requirement.

**Expected behavior**
I would expect the secrets to be listed.

**Environment summary**
$ `cat os-release | grep VERSION=`
VERSION="18.04.2 LTS (Bionic Beaver)"
$ `az --version | grep azure-cli`
azure-cli                         2.0.61
$ `bash --version | grep 'GNU bash'`
GNU bash, version 4.4.19(1)-release (x86_64-pc-linux-gnu)
$ `python3 --version`
Python 3.6.7

**Additional context**
Related to https://github.com/Azure/azure-cli/issues/7311

### PR DESCRIPTION
**Pull request by bim-msft**
---
Fix #8921 

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/_params.py [hunk 1]
```diff
@@ -164,6 +164,10 @@ def load_arguments(self, _):
 
     with self.argument_context('keyvault key set-attributes') as c:
         c.attributes_argument('key', KeyAttributes)
+
+    for scope in ['list', 'list-deleted', 'list-versions']:
+        with self.argument_context('keyvault key {}'.format(scope)) as c:
+            c.argument('maxresults', options_list=['--maxresults'], type=int)
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: is it necessary to set a upper limit on maxresults?

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: @yungezz We have already set the lower bound and upper bound of `maxresults` in swagger and SDK code, it's not necessay to set it again in CLI code.

Swagger code: https://github.com/Azure/azure-rest-api-specs/blob/84ab02cad8cb3d3c97118e989815dbe5c7d5b529/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.0/keyvault.json#L1179-L1189

SDK code: https://github.com/Azure/azure-sdk-for-python/blob/657d9c8e6cbb8bd543d2996112ee9f00dd45cf87/sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/_generated/v7_0/operations/_key_vault_client_operations.py#L1743-L1744

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: What would happen if maxresults passed in is greater than 25?

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: @fengzhou-msft CLI will throw an error message:

```
(env3) D:\repos\azure-cli>az keyvault secret list --vault-name bim-foo --maxresults 100
Parameter 'maxresults' must be less than 25.
```

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: Ok. At least users know the limit from the error message. If you could put it in help text, that will be more clear, just a suggestion though.

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: @fengzhou-msft Yes, I agree. I think it's a better idea that the upcoming CLI codegen system auto-generates such type of help messages from swagger file.

Comment (yonzhan) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py]: Good to know that you will collect all these feedbacks and backpropagate to CodeGen team.

**Pull request merged**